libs/fluid-elements/popover/src/lib/popover.ts

```diff
@@ -0,0 +1,224 @@
+
+/**
+ * Set defaults for preventing overflow
+ * Applied to all instances of the FluidPopover
+ */
+preventOverflow.options = {
+  mainAxis: true,
+  altAxis: true,
+};
+
+type FluidPopoverOffsetFunction = ({
+  popper,
+  reference,
+  placement,
+}: {
+  popper: Rect;
+  reference: Rect;
+  placement: Placement;
+}) => [number | null | undefined, number | null | undefined];
+
+export type FluidPopoverOffset =
+  | FluidPopoverOffsetFunction
+  | [number | null | undefined, number | null | undefined]
+  | undefined;
+
+/**
+ * This is an experimental popover element built with lit-elements and
+ * web-components. It registers itself as `fluid-popover` custom element.
+ * @element fluid-popover
+ * @cssprop --fluid-popover--foreground-key - Controls the foreground color for the key setting.
+ * @cssprop --fluid-popover--background-key - Controls the background color for the key setting.
+ * @cssprop --fluid-popover--background-key - Controls the background color for the key setting.
+ * @cssprop --fluid-popover--foreground-key-hover - Controls the foreground hover color for the key setting.
+ * @cssprop --fluid-popover--background-key-hover - Controls the background hover color for the key setting.
+ * @cssprop --fluid-popover--foreground-disabled - Controls the foreground color for the disabled state.
+ * @cssprop --fluid-popover--background-disabled - Controls the background color for the disabled state.
+ */
+@customElement('fluid-popover')
+export class FluidPopover extends LitElement {
+  /** Styles for the popover component */
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        display: block;
+
+        --fluid-popover--background: var(--color-neutral-50);
+      }
+
+      .fluid-popover {
+        ${unsafeCSS(fluidDtText())};
+        display: none;
+        overflow: scroll;
+        max-height: 350px;
+        background: var(--fluid-popover--background);
+        box-shadow: 0 3px 10px rgba(21, 23, 27, 0.2);
+        transition: opacity 1500ms ease-in-out;
+        z-index: 10;
+      }
+
+      .fluid-popover--open {
+        display: block;
+      }
+    `;
+  }
+
+  /**
+   * Defines whether the popover is open.
+   * @attr
+   * @type boolean
+   * @default false
+   */
+  @property({ type: Boolean, reflect: true })
+  get open(): boolean {
+    return this._open;
+  }
+  set open(value: boolean) {
+    const oldOpen = this._open;
+    this._open = value;
+    this._togglePopover();
+    this.requestUpdate(`open`, oldOpen);
+  }
+  private _open = false;
+
+  /**
+   * The element to connect the popover to
+   * @type HTMLElement
+   */
+  @property({ attribute: false })
+  anchor: HTMLElement;
+
+  /** Defines the placement of the popper.js popover */
+  @property({ type: String, reflect: true })
+  placement: Placement = `bottom-start`;
+
+  /** Defines the placement of the popper.js popover */
+  @property({ type: Array, reflect: true })
+  get fallbackplacement(): Placement[] {
```

Comment:
Not sure how this is for web-components, but normally we use camelCase for properties: `fallbackPlacement`

Reply:
I talked about this with @tomheller, who said that camelCase would be against the standard, referring to the [html attribute reference](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes).
I wanted to camelCase first too, but after the brief discussion I'm definitely on his side 😄